### DISCUSSION TITLE
List of servers not complete

### DISCUSSION THREAD

**Discussion by reinhardwh** [jamulussoftware/jamulus#2593] [Bug Investigation]
**Describe the bug**

List of servers not complete due to blocked fragmented packages

From Facebook:

A little tip I discovered today which might help new users:

had an empty list under genre 1 and 2, no servers showing up. 

But the jamulus explorer website
showed hundreds of servers.
So I did some googling and apparently the problem
was my modem's firewall blocking fragmented packages. 

Turned that option off and now Jamulus seems more alive than I thought it was.
This may have been mentioned somewhere else, but I'm relatively new and just found out about it.
Happy jamming all!

**To Reproduce**

Change setting in router

**Expected behavior**

Comlete list of servers displayed

**Screenshots**

![image](https://user-images.githubusercontent.com/77500932/161485141-8d2e028e-809a-4567-aabb-1e37fbebddc4.png)

**Operating system**

**Version of Jamulus**

**Additional context**

**Comment by jujudusud** [comment]
That's good to know but how can Jamulus replace a network manager?
Jamulus users are masters of their firewall and this can't be considered as a software bug.
It can eventually be included in the documentation but we need to discuss it beforehand to decide the best place.

**Comment by reinhardwh** [comment]
Just wondering why some user encounter "fragmented packages" from Jamulus explorer page.

**Comment by softins** [comment]
> Just wondering why some user encounter "fragmented packages" from Jamulus explorer page.
> 

*Everybody* receives the server list for Any Genre 1 and Any Genre 2 in fragmented packets. And if any other genre became popular enough to have a large list, it would apply there too.

The list of servers is sent as a single UDP datagram, and this is a fundamental legacy of the original design of the Jamulus protocol.

A UDP datagram is first encapsulated by the network stack into an IP packet. The IP packet is then passed to the physical network layer, e.g. Ethernet or some kind of inter-router connection. But before doing that, the IP layer checks the maximum packet size supported by the physical layer. For Ethernet this is 1500 bytes, which includes the Ethernet header. If the IP packet is small enough to fit in a single physical packet, it is just passed down and sent as-is. If not, the IP layer will split it into two or more separate IP fragments that will fit, and will include header information to enable the recipient to reassemble all the fragments into the original IP packet, which is then passed back up to the UDP layer and then to the receiving application.

Because some kinds of denial attacks use IP fragments to exhaust resources on the target host, some routers enable fragments to be blocked. But this prevents the large fragmented Jamumus messages from getting through.

The only way to overcome this is to completely redesign the Jamulus protocol in a non-backwardly-compatible way, and this has never yet become a high enough priority to tackle.

There have been two different attempts in 2020 to mitigate this issue for some users (#582 and #637), but neither addresses the real problem.

**Comment by reinhardwh** [comment]
Would it be possible to write a list of all servers into XML based file and generate one standard web page out of this?

**Comment by softins** [comment]
> Would it be possible to write a list of all servers into XML based file and generate one standard web page out of this?

You could do something like that by processing the output from the Jamulus Explorer back-end (see the Github link for jamulus-php on the Explorer page), but it would just give you something similar to Explorer itself. It isn't something that a Jamulus client could use.

**Comment by reinhardwh** [comment]
Thats a good point & thanks for sharing the insights :)

**Comment by gilgongo** [comment]
Thanks @reinhardwh - Since this isn't a specification for a change, I'm moving to Discussions.
